core/interfaces/driven.go

```diff
@@ -84,6 +84,9 @@ type Storage interface {
 	DeleteLeaderboardEntries(leaderboardID string, orgID string, appID string, leavingUserIDs []string) error
 	DeleteAllLeaderboardEntries(leaderboardID string, orgID string, appID string) error
 	InitLeaderboardEntryScores(orgID string, appID string) error
+
+	FindScores(filter interface{}, limit *int, offset *int) ([]model.Score, error)
```

Comment:
After you move your code that has mongo dependencies into the `storage` package, these interface function definitions will need to be updated.